File: results-05-12/python/python-00010-benchmark-summarize-website/qwen2-5-coder-32b/editor/final-src-code.py

import os
import json
os.environ["SHINKAI_NODE_LOCATION"] = "http://localhost:9950"
os.environ["BEARER"] = "debug"
os.environ["X_SHINKAI_TOOL_ID"] = "tool-id-debug"
os.environ["X_SHINKAI_APP_ID"] = "tool-app-debug"
os.environ["X_SHINKAI_LLM_PROVIDER"] = "o_qwen2_5_coder_32b"
os.environ["HOME"] = "results/python/python-00010-benchmark-summarize-website/qwen2-5-coder-32b/editor/home"
os.environ["MOUNT"] = "results/python/python-00010-benchmark-summarize-website/qwen2-5-coder-32b/editor/mount"
os.environ["ASSETS"] = "results/python/python-00010-benchmark-summarize-website/qwen2-5-coder-32b/editor/assets"

from typing import Dict, Any
from shinkai_local_tools import benchmark_store_website, shinkai_llm_prompt_processor

class CONFIG:
    pass

class INPUTS:
    url: str

class OUTPUT:
    summary: str

async def run(config: CONFIG, inputs: INPUTS) -> OUTPUT:
    # Fetch and store website content
    website_data = benchmark_store_website(inputs.url)
    
    # Generate a summary using LLM prompt processor
    prompt = f"Please provide a concise summary of the following webpage content:\n{website_data}"
    response = shinkai_llm_prompt_processor(format='text', prompt=prompt)
    
    output = OUTPUT()
    output.summary = response['message']
    return output

if __name__ == "__main__":
    import asyncio
    
    config = CONFIG()
    inputs = INPUTS()
    inputs.url="https://raw.githubusercontent.com/acedward/expert-octo-computing-machine/main/test.html"
    
    # Run the async function
    result = asyncio.run(run(config, inputs))
    print(json.dumps(result.__dict__))

File: results-05-12/typescript/typescript-00021-benchmark-summarize-website/qwen2-5-coder-32b/editor/final-src-code.ts

  // These environment variables are required, before any import.
  // Do not remove them, as they set environment variables for the Shinkai Tools.
  Deno.env.set('SHINKAI_NODE_LOCATION', "http://localhost:9950");
  Deno.env.set('BEARER', "debug");
  Deno.env.set('X_SHINKAI_TOOL_ID', "tool-id-debug");
  Deno.env.set('X_SHINKAI_APP_ID', "tool-app-debug");
  Deno.env.set('X_SHINKAI_LLM_PROVIDER', "o_qwen2_5_coder_32b");
  Deno.env.set('HOME', "results/typescript/typescript-00021-benchmark-summarize-website/qwen2-5-coder-32b/editor/home");
  Deno.env.set('MOUNT', "results/typescript/typescript-00021-benchmark-summarize-website/qwen2-5-coder-32b/editor/mount");
  Deno.env.set('ASSETS', "results/typescript/typescript-00021-benchmark-summarize-website/qwen2-5-coder-32b/editor/assets");
  
import { shinkaiSqliteQueryExecutor, shinkaiLlmPromptProcessor } from './shinkai-local-tools.ts';

type CONFIG = {};
type INPUTS = { url: string };
type OUTPUT = { summary: string };

export async function run(config: CONFIG, inputs: INPUTS): Promise<OUTPUT> {
    const { url } = inputs;

    // Read the website content from the SQLite database
    const query = `SELECT content FROM websites WHERE url = ?`;
    const result = await shinkaiSqliteQueryExecutor('default', query, [url]);

    if (result.rowCount === 0) {
        throw new Error(`No content found for URL: ${url}`);
    }

    const content = result.result[0].content;

    // Generate a summary using the AI LLM
    const prompt = `Please provide a summary of the following text:\n\n${content}`;
    const summaryResult = await shinkaiLlmPromptProcessor('text', prompt);

    return {
        summary: summaryResult.message,
    };
}

  
  // console.log('Running...')
  // console.log('Config: {}')
  // console.log('Inputs: {"url":"https://raw.githubusercontent.com/acedward/expert-octo-computing-machine/main/test.html"}')
  
  try {
    const program_result = await run({}, {"url":"https://raw.githubusercontent.com/acedward/expert-octo-computing-machine/main/test.html"});
    if (program_result) console.log(JSON.stringify(program_result, null, 2));
    else console.log(program_result);
  } catch (e) {
    console.log('::ERROR::', e);
  }

